docs/dg/dev/backend-development/data-manipulation/create-an-order-management-system-spryker-commerce-os.md

```diff
@@ -413,8 +413,146 @@ You can keep moving the item until the order is closed.
+
+Default commands and conditions:
+- `\SprykerTest\Zed\Oms\Helper\Mock\AlwaysTrueConditionPluginMock`: condition that always returns true
+- `\SprykerTest\Zed\Oms\Helper\Mock\AlwaysFalseConditionPluginMock`: condition that always returns true
+- `\SprykerTest\Zed\Oms\Helper\Mock\CommandByItemPluginMock`: a mock for `CommandByItemInterface` that always returns an empty array
+- `\SprykerTest\Zed\Oms\Helper\Mock\CommandByOrderPluginMock`: a mock for `CommandByOrderInterface` that always returns an empty array
+
+
+
+{% info_block infoBox %}
+
+* Commands and conditions can't be defined at runtime during tests. Instead, they must be specified in `codeception.yml`.
+* Test scenarios support only a single item.
+* Timeouts can't be tested.
+
+{% endinfo_block %}
+
+#### 5.2 Create a test for your state machine
+
+Test example:
+
+```php
+<?php
+
+declare(strict_types = 1);
+
+namespace PyzTest\Zed\YourModuleName\Integration\Oms;
+
+use Codeception\Test\Unit;
+use PyzTest\Zed\YourModuleName\YourModuleNameIntegrationTester;
+
+class OmsIntegrationTest extends Unit
+{
+	/**
+     * @var string
+     */
+    protected const STATE_MACHINE_NAME = 'ForeignPaymentStateMachine01';
+
+    /**
+     * @param \Codeception\TestInterface $test
+     *
+     * @return void
+     */
+    public function _before(TestInterface $test): void
+    {
+        parent::_before($test);
+
+        $xmlFileDirectory = APPLICATION_VENDOR_DIR . 'spryker/spryker/Bundles/SalesPayment/config/Zed/Oms/';
+        $this->getSalesOmsHelper()->setupStateMachine(static::STATE_MACHINE_NAME, $xmlFileDirectory);
+    }
+
+	public function testMoveAnItemFromStateAToStateB(): void
+    {
+    	// Set up one single item with its current expected state
+        $this->tester->haveOrderItemInState('a');
+
+		// do something where you expect that the item should move to the next state
+
+		// Trigger the state machine
+        $this->tester->tryToTransitionOrderItems();
+        // or
+        $this->tester->tryToTransitionOrderItems('event name');
+
+        // Assert that the item is moved to the expected state.
+        $this->tester->assertOrderItemIsInState('b');
+    }
+}
+```
+
+The state machine is set up in the `_before` method. You need to specify the state machine name and the path to the XML files containing your definitions.
+
+1. The `haveOrderItemInState` method initializes an item in `a` state. This method also accepts a second argument, allowing you to pass custom fields for order item creation. After setting up the item, add the logic that should move it to the next state, such as receiving an async API message or calling a command.  
+
+2. The `tryToTransitionOrderItems` method (use only one of these methods) attempts to transition the order item to the next state. When used without an event name, it only checks conditions. To trigger a specific event, pass the event name as a parameter.
+
+3. The last method asserts that the item is in the expected `b` state.
```

Comment:
What's the method? They're specified for prev steps

Reply:
Added